analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -438,10 +438,15 @@ class Pip(
             it.isBlank() || it == "UNKNOWN"
         }
 
-    private fun getLicenseFromClassifier(classifier: String): String? =
+    private fun getLicenseFromClassifier(classifier: String): String? {
         // Example license classifier:
         // "License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)"
-        classifier.split(" :: ").takeIf { it.first() == "License" }?.last()?.takeUnless { it == "OSI Approved" }
+        return if (classifier.startsWith("License :: OSI Approved :: ")) {
+             classifier.substringAfterLast(" :: ", missingDelimiterValue = "").trim().ifEmpty { null }
+        } else {
+            null
+        }
+    }
```

Comment:
I'd propose instead:

    private fun getLicenseFromClassifier(classifier: String): String? {
        // Example license classifier:
        // "License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)"
        val classifiers = classifier.split(" :: ").map { it.trim() }
        val license = classifiers.takeIf { it.first() == "License" }?.last()
        return license?.takeUnless { it in listOf("License", "OSI Approved") }
    }